lib/mpl_toolkits/mplot3d/axis3d.py

```diff
@@ -371,30 +354,25 @@ def draw(self, renderer):
         self.offsetText.draw(renderer)
 
         # Draw grid lines
-        if len(xyz0) > 0:
+        if self.axes._draw_grid and len(ticks):
             # Grid points at end of one plane
-            xyz1 = copy.deepcopy(xyz0)
+            xyz1 = xyz0.copy()
             newindex = (index + 1) % 3
             newval = get_flip_min_max(xyz1[0], newindex, mins, maxs)
-            for i in range(len(majorLocs)):
-                xyz1[i][newindex] = newval
+            xyz1[:, newindex] = newval
 
             # Grid points at end of the other plane
-            xyz2 = copy.deepcopy(xyz0)
+            xyz2 = xyz0.copy()
             newindex = (index + 2) % 3
             newval = get_flip_min_max(xyz2[0], newindex, mins, maxs)
-            for i in range(len(majorLocs)):
-                xyz2[i][newindex] = newval
-
-            lines = list(zip(xyz1, xyz0, xyz2))
-            if self.axes._draw_grid:
-                self.gridlines.set_segments(lines)
-                self.gridlines.set_color([info['grid']['color']] * len(lines))
-                self.gridlines.set_linewidth(
-                    [info['grid']['linewidth']] * len(lines))
-                self.gridlines.set_linestyle(
-                    [info['grid']['linestyle']] * len(lines))
-                self.gridlines.draw(renderer, project=True)
+            xyz2[:, newindex] = newval
+
+            lines = np.stack([xyz1, xyz0, xyz2], axis=1)
+            self.gridlines.set_segments(lines)
+            self.gridlines.set_color(info['grid']['color'])
+            self.gridlines.set_linewidth(info['grid']['linewidth'])
+            self.gridlines.set_linestyle(info['grid']['linestyle'])
```

Comment:
do we know if these setters are properly broadcasting the scalars?

Reply:
At least manually setting a scalar into any of these works.